### PR TITLE
Updated Vanilla Vehicles Expanded/added Tier 3 support

### PR DESCRIPTION
**Pull request by SokyranTheDragon**
Anything from VVE Tier 3 that needed patching was included in the original vehicle mod (despite it not using it), so it acts as a framework. No need for separate patch for Tier 3 assembly, as it barely does anything (and nothing that needs patching).

Added temporary sync workers for Designation and DesignationManager, simplifying existing patch and one of the new ones.

Currently, one of the vehicles (Bunsen) causes desyncs when using its turrets. Currently investigating it, but seems like it _may_ be an issue with Vanilla Expanded Framework. Not 100% sure yet.

### CODE REVIEW

File: Source/MpSyncWorkers.cs [hunk 1]
```diff
@@ -63,6 +70,43 @@ private static void SyncGameCondition(SyncWorker sync, ref GameCondition gameCon
...
+            {
+                var manager = sync.Read<DesignationManager>();
+
+                if (manager != null)
+                {
+                    var target = sync.Read<LocalTargetInfo>();
+                    var def = sync.Read<DesignationDef>();
+
+                    if (target.HasThing)
+                        manager.DesignationOn(target.Thing, def);
```
Comment (Zetrith) [inline Source/MpSyncWorkers.cs:95]: Shouldn't this assign to designation?

Comment (SokyranTheDragon) [reply Source/MpSyncWorkers.cs]: Good catch, should be fixed.

I've had to re-make this code as I seemed to have lost the original, so I must have messed up when doing this.

**Comment by SokyranTheDragon** [comment]
Just to add to the last point:

- The issue with Bunsen only happens when there's both a Workshop and a local copy of Vehicle Framework, while specifically running the Workshop one
- The issue will likely happen with other mods though, so it's something that needs to be fixed anyway

**Comment by Zetrith** [comment]
I'll merge but the fact the new SyncWorker affects game state and isn't a pure data transform should be looked into.

**Comment by Zetrith** [comment]
Ah, I was sure `DesignationOn` added a new one. All good then.

**Pull request merged**